### DISCUSSION TITLE
Best way to stop and restart reading I2S from microphone

### DISCUSSION THREAD

**Discussion by biomurph** [pschatzmann/arduino-audio-tools#1827] [Q&A] [answered]
I am using an SPH0645 microphone (Adafruit Breakout) and a XIAO ESP32S6.
Eventually, I will stream the audio to a networked computer, but I’m just getting started and I have dug around to figure out the best way to ‘stop’ and ‘start’ the I2C. When I am not connected to any clients, I want to stop clocking the microphone.
The only thing I could find in the library was an end() method. Is this the best way? Does it destroy the original setup and have to be re-made as in the setup? I am configuring pins and sample rate and such. Do I have to do all that again when I want to turn it back on?

**Comment by pschatzmann** [comment, marked as answer]
I think you mean I2S and not I2C. Calling end() is shutting down the I2S all together and calling begin() will restart it with the last defined settings (or default values when nothing has been defined before).

Alternatively you can start the system from processing any data by just stop calling copy(), or you could call end() on the StreamCopy copier which has the same effect. In this case I2S will still be active and will consume some power, but it will be quicker to stop and resume the processing.

**Comment by biomurph** [comment]
Thanks for the quick reply!
Yes, I meant I2S. 

I will use the end() and begin() and see how that goes.
When I do end the I2S, do I have to stop calling the copy() function too? I imagine it would just send zeros or something if there was no I2S running?
